### PR TITLE
gh-118820: Zero-valued flag enum has no name

### PR DESCRIPTION
**Pull request by nineteendo**
* Issue: gh-118820

----
📚 Documentation preview 📚: https://cpython-previews--118848.org.readthedocs.build/

### CODE REVIEW

File: Doc/howto/enum.rst [hunk 1]
```diff
@@ -1149,6 +1154,8 @@ the following are true:
 
     >>> (Color.RED | Color.GREEN).name
     'RED|GREEN'
+    >>> repr(Perm.R & Perm.W).name)
+    'None'
```
Comment (ethanfurman) [inline Doc/howto/enum.rst]:
```suggestion
    >>> (Perm.R & Perm.W).name is None
    True
```

Also, change `snippet` above to `snippets`.

**Review by ethanfurman — CHANGES_REQUESTED** [review]

**Comment by nineteendo** [comment]
ethanfurman, I have made the requested changes; please review again.

**Review by ethanfurman — APPROVED** [review]

**Comment by TeamSpen210** [comment]
It might be more clear to do `Perm(0)`, since it took me a moment to realise the intersection would exclude everything.

**Comment by nineteendo** [comment]
Note that `...(0)` is only used once on the page, `... & ...` is used 4 more times. But if Ethan agrees, I'll change it.

File: Doc/howto/enum.rst [hunk 2]
```diff
@@ -1149,6 +1154,8 @@ the following are true:
 
     >>> (Color.RED | Color.GREEN).name
     'RED|GREEN'
+    >>> (Perm.R & Perm.W).name is None
```
Comment (ethanfurman) [inline Doc/howto/enum.rst]:
```suggestion
    >>> (Perm.R & Perm.W).name is None                   # effectively `Perm(0)`
```

**Review by ethanfurman — CHANGES_REQUESTED** [review]

File: Doc/howto/enum.rst (hunk 2 above)
Comment (ethanfurman) [reply Doc/howto/enum.rst]: Let's add the `Perm(0)` as a comment on the same line.  Thank you, both.

**Comment by nineteendo** [comment]
I have made the requested changes; please review again.

**Review by ethanfurman — APPROVED** [review]

File: Doc/howto/enum.rst [hunk 3]
```diff
@@ -1149,6 +1154,8 @@ the following are true:
 
     >>> (Color.RED | Color.GREEN).name
     'RED|GREEN'
+    >>> (Perm.R & Perm.W).name is None                   # effectively `Perm(0)`
```
Comment (picnixz) [inline Doc/howto/enum.rst]:
```suggestion
    >>> (Perm.R & Perm.W).name is None  # effectively Perm(0)
```

I don't think we need that much space (unless it is perhaps aligned with something I don't see?) In addition, I wouldn't use single-quote references in the doctest. Finally, maybe you could actually include the Perm snippet here only, and not at the beginning (you are only using it for this specific example).

Comment (nineteendo) [reply Doc/howto/enum.rst]: I think it was mostly to move it out of the way. Moved declaration to the example.

**Review by ethanfurman — APPROVED** [review]

**Comment by nineteendo** [comment]
Thanks Ethan.

**Comment by nineteendo** [comment]
Do we backport this? Or can we close the issue immediately?

**Comment by ethanfurman** [comment]
No problem, apologies for the delay.  Should probably backport.  Adding appropriate labels.

**Pull request merged**